pkg/arm/data/azuredeploy.json

```diff
@@ -80,11 +81,12 @@
                 ]
             }
         },
+{{ if eq ( index .ContainerService.Properties.AgentPoolProfiles 0).VnetSubnetID $.Config.DefaultVnetSubnetID }}              
```

Comment:
for now, the PR you're based on will guarantee that there are 2 agent pool profiles and that the vnetsubnetID is identical across the two.
you can decide to create a vnet based on `{{ if not (index .ContainerService.Properties.AgentPoolProfiles 0).VnetSubnetID }}`